### PR TITLE
Upgrade to Pyodide 0.18.0

### PR DESCRIPTION
**Pull request by bollwyvl**
## References

- [x] follow-on / contains #178
  - got a bit confused with the history, and didn't want to push against the main repo branch :blush:
- [ ] blocked by #254
  - really need to get some of the heavy hitters like `matplotlib` out of the critical load path for interactive testing!

## Code changes

- [x] explicit call to install `micropip`
- [x] make `copy`s of long-running python objects like `kernel`, `interpreter` and `*_stream`s to avoid GC/ref errors 
- [x] update outer environment(s) to python 3.9 (as pyodide is now _also_ 3.9)
  - [ ] add a `matrix` for `python-version` in CI to ensure we maintain compat back to advertised `3.7` (this could be a separate PR)
- [x] **remove** debug messages for determining current pyodide `dev` API support for `fileSystem`/`FS`
- [ ] add a `py - filesystem.ipynb` with some interactive examples of this API
  - [x] show using `pyodide_js.FS` with stdlib python, e.g. `pathlib` 
  - [x] show using `pyodide_js.FS.filesystems.IDBFS` directly 
  - [ ] contribute back to `pyodide` upstream

## User-facing changes

- n/a  (yet, though we'd like to see some performance benefits)
 
## Backwards-incompatible changes

- going to 3.9 shouldn't be too big of an issue... but probably non-negotiable!

## Notes

- some work-in-progress notebook stuff, showing how user python code will be able to interact with `FS`:
  - https://gist.github.com/bollwyvl/616802a1765dc702ee97a0f065adad8c

### CODE REVIEW

**Comment by jtpio** [comment]
Nice, thanks for starting this!

>     * really need to get some of the heavy hitters like `matplotlib` out of the critical load path for interactive testing!

Maybe it will be possible to bring `importhook` back with the latest Pyodide if it changed something regarding the recursion limit.

**Comment by bollwyvl** [comment]
An update: it looks like the nightlies with `FS` are up... but `Pygments.js` is not loading correctly, so still blocked for a tick...

**Comment by bollwyvl** [comment]
Ok, I've moved that notebook from the gist into this PR for the time being, so that it shows up under the docs build. The example now includes persisting to `IndexedDB` from user python code, which is a little crazy. Certainly a lot left to do, but getting to a live interactive state is what this is all about, right? Here's hoping i didn't mess up the docs build or something :crossed_fingers: .

File: packages/pyolite-kernel-extension/src/index.ts [hunk 1]
```diff
@@ -12,7 +12,7 @@ import { PyoliteKernel } from '@jupyterlite/pyolite-kernel';
 /**
  * The default CDN fallback for Pyodide
  */
-const PYODIDE_CDN_URL = 'https://cdn.jsdelivr.net/pyodide/v0.17.0/full/pyodide.js';
+const PYODIDE_CDN_URL = 'https://cdn.jsdelivr.net/pyodide/dev/full/pyodide.js';
```
Comment (jtpio) [inline packages/pyolite-kernel-extension/src/index.ts:15]: If there was a way to pin on a commit / hash, maybe we could already start shipping it.

Comment (bollwyvl) [reply packages/pyolite-kernel-extension/src/index.ts:15]: ha, would rather not, just yet... i think waiting for the npm releases will be worth it...

**Comment by bollwyvl** [comment]
SNAAAP:

https://github.com/pyodide/pyodide/releases/tag/0.18.0

**Comment by bollwyvl** [comment]
It appears we semi-frequently get into trouble with our reference to `kernel` in the worker

```
Uncaught (in promise) Error: This borrowed attribute proxy was automatically destroyed in the process of destroying the proxy it was borrowed from. Try using the 'copy' method.
    at _getPtr (pyodide.js:1)
    at pyodide.js:1
    at Object.set (pyodide.js:1)
    at self.onmessage (75af6e0a-7e2c-4e54-ad63-8249388b1216:315)
```

**Comment by bollwyvl** [comment]
Was able to locally run through all of the examples except `folium` (which needs a lot of hacks _anyway_, apparently).

**Comment by bollwyvl** [comment]
On RTD, it looks like all of the cruft from `folium` can be boiled down to:

```
import micropip
await micropip.install("folium==0.12.1")
```

MarkupSafe is now available directly from `pyodide` as a pre-built wasm package.

**Comment by bollwyvl** [comment]
The only hard regression appears to be a `maximum call stack` error in `ipyleaflet`: using the download technique employed by [micropip](https://github.com/pyodide/pyodide/blob/0.18.0/packages/micropip/src/micropip/micropip.py#L60), was able to get around it...

**Comment by bollwyvl** [comment]
Well I guess we need to merge this in order to start looking into some of the other open PRs and issues. Hold on to your hats!

**Pull request merged**